tests/infra/consortium.py

```diff
@@ -324,14 +325,43 @@ def get_proposal(self, remote_node, proposal_id):
             assert r.status_code == http.HTTPStatus.OK.value
             return r.body.json()
 
-    def retire_node(self, remote_node, node_to_retire, timeout=10):
+    def retire_node(self, remote_node, node_to_retire, timeout=10, network=None):
+        pending = False
+        with remote_node.client(connection_timeout=timeout) as c:
+            r = c.get(f"/node/network/nodes/{node_to_retire.node_id}")
+            pending = r.body.json()["status"] == infra.node.State.PENDING.value
         LOG.info(f"Retiring node {node_to_retire.local_node_id}")
         proposal_body, careful_vote = self.make_proposal(
             "remove_node",
             node_id=node_to_retire.node_id,
         )
         proposal = self.get_any_active_member().propose(remote_node, proposal_body)
         self.vote_using_majority(remote_node, proposal, careful_vote)
+        if remote_node == node_to_retire:
+            remote_node, _ = network.wait_for_new_primary(remote_node)
+        if remote_node.version_after("ccf-2.0.4") and not pending:
+            end_time = time.time() + timeout
+            r = None
+            while time.time() < end_time:
+                try:
+                    with remote_node.client(connection_timeout=timeout) as c:
+                        r = c.get("/node/network/removable_nodes").body.json()
+                        if node_to_retire.node_id in {n["node_id"] for n in r["nodes"]}:
+                            check_commit = infra.checker.Checker(c)
+                            r = c.delete(
+                                f"/node/network/nodes/{node_to_retire.node_id}"
+                            )
+                            check_commit(r)
+                            break
+                        else:
+                            r = c.get(
```

Comment:
`r` is unused here?

Reply:
Mostly yes, but if we end up raising at :364, it's nice to know what the last state was (ie. what we're stuck on), and that's why this line is there at all.